### ISSUE TITLE
cex.io trade history is broken

### ISSUE THREAD

**Issue by npomfret** [knowm/XChange#2623]
(Here as a reminder - for me because I'm at fault for at least some of this!)  But the cex.io user trades parsing is not accurate, the api is way more complicated than expected.

**Comment by wvr** [comment]
Got in a bit late for the 4.3.7 release, but saw the problem and PR'd it: https://github.com/knowm/XChange/pull/2632

The problem was with having cancelled orders being returned by the API, they don't have a counter currency amount, so resulted in an NPE in the price calculation.

Agreed, the API is a wee bit more verbose than the usual.

**Comment by npomfret** [comment]
@timmolter please reopen.  I think the problems go deeper than the NPE described.  I've not got it figured yet, but market order v's limit orders are modelled differently, as as buys and sells and partially cancelled.

**Comment by wvr** [comment]
Agreed on re-open and revisit. I've not tested with market orders as I actively avoid those, only did a quick-fix to handle cancelled orders as well (as filled and partially filled). 

I've not delved too deeply into that related regex work of yours @npomfret. Makes sense to me, but I prefer to steer well clear of those.
